src/bio/bio_recovery.c

```diff
@@ -223,9 +243,35 @@ setup_xs_bs(void *arg)
 			D_ERROR("Failed to create io channel for %p\n", bbs);
 			return;
 		}
-		/* Blobs (VOS pools) will be opened in reint reaction */
+	}
+
+	/* If reint will be tirggered later, blobs will be opened in reint reaction */
+	if (bbs->bb_dev->bb_trigger_reint) {
 		D_ASSERT(d_list_empty(&bxb->bxb_io_ctxts));
+		goto done;
+	}
+
+	/* Open all blobs when reint won't be tirggered */
+	d_list_for_each_entry(ioc, &bxb->bxb_io_ctxts, bic_link) {
+		if (ioc->bic_blob != NULL && !ioc->bic_closing)
+			continue;
+
+		closed_blobs += 1;
+		if (ioc->bic_opening || ioc->bic_closing)
+			continue;
+
+		D_ASSERT(ioc->bic_blob_id != SPDK_BLOBID_INVALID);
+		/* device type and flags will be ignored in bio_blob_open() */
+		bio_blob_open(ioc, true, 0, SMD_DEV_TYPE_MAX, ioc->bic_blob_id);
+	}
+
+	if (closed_blobs) {
+		D_DEBUG(DB_MGMT, "blobstore:%p has %d closed blobs\n",
```

Comment:
maybe a assertion D_ASSERT(bxb->bxb_ready == true).

Reply:
why? the bxb_ready should be false in this case.